src/Sylius/Component/Channel/spec/Context/CachedPerRequestChannelContextSpec.php

```diff
@@ -116,7 +115,30 @@ function it_does_not_cache_channel_not_found_exceptions_for_null_master_requests
 
         $decoratedChannelContext
             ->getChannel()
-            ->will(CompositePromise::it()->willThrow(ChannelNotFoundException::class)->andThenReturn($channel))
+            ->will(new class ($channel->getWrappedObject()) {
+                /** @var int */
+                private $counter = 0;
+
+                /** @var ChannelInterface */
+                private $channel;
+
+                public function __construct(ChannelInterface $channel) {
+                    $this->channel = $channel;
+                }
+
+                /** @throws ChannelNotFoundException */
+                public function __invoke(): ChannelInterface
+                {
+                    $currentCounter = $this->counter;
+                    ++$this->counter;
+
+                    if ($currentCounter === 0) {
```

Comment:
Yeah I know it could be made smarter by:

```suggestion
                    if ($this->counter++ === 0) {
```

But it feels better to have more descriptive code.